sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -215,3 +215,56 @@ case class GpuSize(child: Expression, legacySizeOfNull: Boolean)
     }
   }
 }
+
+case class GpuSortArray(base: Expression, ascendingOrder: Expression)
+    extends GpuBinaryExpression with ExpectsInputTypes {
+
+  override def left: Expression = base
+
+  override def right: Expression = ascendingOrder
+
+  override def dataType: DataType = base.dataType
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(ArrayType, BooleanType)
+
+  override def checkInputDataTypes(): TypeCheckResult = base.dataType match {
+    case ArrayType(dt, _) if RowOrdering.isOrderable(dt) =>
+      ascendingOrder match {
+        // replace Literal with GpuLiteral here
+        case GpuLiteral(_: Boolean, BooleanType) =>
+          TypeCheckResult.TypeCheckSuccess
+        case _ =>
+          TypeCheckResult.TypeCheckFailure(
+            "Sort order in second argument requires a boolean literal.")
+      }
+    case ArrayType(dt, _) =>
+      val dtSimple = dt.catalogString
+      TypeCheckResult.TypeCheckFailure(
+        s"$prettyName does not support sorting array of type $dtSimple which is not orderable")
+    case _ =>
+      TypeCheckResult.TypeCheckFailure(s"$prettyName only supports array input.")
+  }
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector =
+    throw new IllegalStateException("This is not supported")
+
+  override def doColumnar(lhs: GpuScalar, rhs: GpuColumnVector): ColumnVector =
+    throw new IllegalStateException("This is not supported")
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {
+    val isDescending = isDescendingOrder(rhs)
+    lhs.getBase.listSortRows(isDescending, true)
+  }
+
+  override def doColumnar(numRows: Int, lhs: GpuScalar, rhs: GpuScalar): ColumnVector = {
```

Comment:
so we can end up here if the expressions we are working with evaluate to scalars? Could you add a test that shows this use case? I could be missing it from the integration test.

Reply:
Yes, I made up the test case for scalar.